c/datasetjson.c

```diff
@@ -1185,6 +1185,52 @@ void deleteDatasetOrMember(HttpResponse* response, char* absolutePath) {
     }  
   }
   else {
+    char *typesArg = defaultDatasetTypesAllowed;
+    int datasetTypeCount = (typesArg == NULL) ? 3 : strlen(typesArg);
+    int workAreaSizeArg = 0;
+    int fieldCount = defaultCSIFieldCount;
+    char **csiFields = defaultCSIFields;
+
+    csi_parmblock *returnParms = (csi_parmblock*)safeMalloc(sizeof(csi_parmblock),"CSI ParmBlock");
+
+    char dsName[45];
+    for (int i = 0; i < sizeof(datasetName.value); i++) {
```

Comment:
Make this a separate function. I really have no idea what the purpose of this is by looking at it, but it looks like it happens more than once.